packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -201,7 +203,11 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         super();
 
         this.logger = ChildLogger.create(
-            dataStoreContext.containerRuntime.logger, undefined, {all:{ dataStoreId: uuid() }});
+            // back-compat 0.38 - logger is added to IFluidDataStoreContext in 0.38.
+            dataStoreContext.logger ?? dataStoreContext.containerRuntime.logger,
```

Comment:
i think we should push this up to the context, and have the context create/own the childlogger for the data store